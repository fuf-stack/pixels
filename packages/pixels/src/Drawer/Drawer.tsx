import type { TVClassName, TVProps } from '@fuf-stack/pixel-utils';
import type { DrawerProps as HeroDrawerProps } from '@heroui/drawer';
import type { ReactNode } from 'react';

import {
  Drawer as HeroDrawer,
  DrawerBody as HeroDrawerBody,
  DrawerContent as HeroDrawerContent,
  DrawerFooter as HeroDrawerFooter,
  DrawerHeader as HeroDrawerHeader,
} from '@heroui/drawer';

import { tv, variantsToClassNames } from '@fuf-stack/pixel-utils';

// drawer variants
export const drawerVariants = tv({
  slots: {
    wrapper: '',
    base: '',
    backdrop: '',
    header: '',
    body: '',
    footer: '',
    closeButton: '',
  },
});

export const drawerSizes = [
  'xs',
  'sm',
  'md',
  'lg',
  'xl',
  '2xl',
  '3xl',
  '4xl',
  '5xl',
  'full',
] as const;
export type DrawerSizes = (typeof drawerSizes)[number];

export const drawerRadii = ['none', 'sm', 'md', 'lg'] as const;
export type DrawerRadius = (typeof drawerRadii)[number];

export const drawerBackdrops = ['opaque', 'transparent', 'blur'] as const;
export type DrawerBackdrop = (typeof drawerBackdrops)[number];

export const drawerPlacements = ['left', 'right', 'top', 'bottom'] as const;
export type DrawerPlacement = (typeof drawerPlacements)[number];

type VariantProps = TVProps<typeof drawerVariants>;
type ClassName = TVClassName<typeof drawerVariants>;

export interface DrawerProps extends VariantProps {
  /** The backdrop style of the Drawer. */
  backdrop?: DrawerBackdrop;
  /** Drawer body content */
  children?: ReactNode;
  /** CSS class name */
  className?: ClassName;
  /** Custom close button to display on top right corner. */
  closeButton?: ReactNode;
  /** Footer of the Drawer. */
  footer?: ReactNode;
  /** Header of the Drawer. */
  header?: ReactNode;
  /** Whether to hide the drawer close button. */
  hideCloseButton?: boolean;
  /** Whether the drawer is open by default (controlled). */
  isOpen?: boolean;
  /** Whether the drawer can be closed by clicking on the overlay or pressing the Esc key. */
  isDismissable?: boolean;
  /** Handler that is called when the drawer is closed. */
  onClose?: () => void;
  /** Handler that is called when the drawer's open state changes. */
  onOpenChange?: (isOpen: boolean) => void;
  /** The drawer position. */
  placement?: DrawerPlacement;
  /** The container element in which the overlay portal will be placed */
  portalContainer?: HeroDrawerProps['portalContainer'];
  /** The drawer border radius. */
  radius?: DrawerRadius;
  /** The drawer size. */
  size?: DrawerSizes;
  /** HTML data-testid attribute used in e2e tests */
  testId?: string;
}

/**
 * Drawer component based on [HeroUI Drawer](https://www.heroui.com//docs/components/drawer)
 */
const Drawer = ({
  backdrop = 'opaque',
  children = null,
  className = undefined,
  closeButton = undefined,
  footer = undefined,
  header = undefined,
  hideCloseButton = false,
  isOpen = undefined,
  isDismissable = true,
  onClose = undefined,
  onOpenChange = undefined,
  placement = 'right',
  portalContainer = undefined,
  radius = 'lg',
  size = 'md',
  testId = undefined,
}: DrawerProps) => {
  // classNames from slots
  const variants = drawerVariants();
  const classNames = variantsToClassNames(variants, className, 'base');

  return (
    <HeroDrawer
      backdrop={backdrop}
      classNames={classNames}
      closeButton={closeButton}
      data-testid={testId}
      hideCloseButton={hideCloseButton}
      isDismissable={isDismissable}
      isOpen={isOpen}
      onClose={onClose}
      onOpenChange={onOpenChange}
      placement={placement}
      portalContainer={portalContainer}
      radius={radius}
      scrollBehavior="inside"
      size={size}
    >
      <HeroDrawerContent data-testid={testId ? `drawer_${testId}` : 'drawer'}>
        {() => (
          <>
            {header && <HeroDrawerHeader>{header}</HeroDrawerHeader>}
            <HeroDrawerBody id="drawer_body">{children}</HeroDrawerBody>
            {footer && <HeroDrawerFooter>{footer}</HeroDrawerFooter>}
          </>
        )}
      </HeroDrawerContent>
    </HeroDrawer>
  );
};

export default Drawer;
