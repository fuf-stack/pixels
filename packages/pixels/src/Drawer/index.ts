import Drawer, {
  drawerBackdrops,
  drawerPlacements,
  drawerRadii,
  drawerSizes,
  drawerVariants,
} from './Drawer';

export type { DrawerProps } from './Drawer';

export {
  Drawer,
  drawerBackdrops,
  drawerPlacements,
  drawerRadii,
  drawerSizes,
  drawerVariants,
};

export default Drawer;
