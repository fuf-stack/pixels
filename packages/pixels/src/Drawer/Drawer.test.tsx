import { describe, expect, test } from 'vitest';

import storySnapshots from '@repo/storybook-config/story-snapshots';
import { render, screen } from '@testing-library/react';

import Drawer from './Drawer';
import * as stories from './Drawer.stories';

describe('Story Snapshots', () => {
  storySnapshots(stories);
});

describe('Drawer Component', () => {
  test('should render with no children', () => {
    const { container } = render(<Drawer />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test('Should have correct testIds', () => {
    const { container } = render(<Drawer testId="drawer1" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test('should render with only header', () => {
    const { container } = render(
      <Drawer
        header={<div data-testid="drawer-header">test</div>}
        testId="drawerTestId"
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  test('should render header, body, and footer when provided', () => {
    render(
      <Drawer
        isOpen // Must be open to render content
        footer={<div data-testid="drawer-footer">Footer Content</div>}
        onClose={() => {}}
        header={<div data-testid="drawer-header">Header Content</div>}
        // eslint-disable-next-line react/no-children-prop
        children={<div data-testid="drawer-body">Body Content</div>}
      />,
    );
    expect(screen.getByTestId('drawer-header')).toHaveTextContent(
      'Header Content',
    );
    expect(screen.getByTestId('drawer-body')).toHaveTextContent('Body Content');
    expect(screen.getByTestId('drawer-footer')).toHaveTextContent(
      'Footer Content',
    );
  });
  test('should not render header, footer if not provided', () => {
    render(
      <Drawer isOpen onClose={() => {}}>
        <div data-testid="drawer-body">Body Content</div>
      </Drawer>,
    );
    // Check if header and footer are NOT in the document
    expect(screen.queryByRole('heading')).not.toBeInTheDocument(); // Assuming HeroDrawerHeader renders a heading
    expect(screen.queryByTestId('drawer-footer')).not.toBeInTheDocument();
    // Body should still be present
    expect(screen.getByTestId('drawer-body')).toHaveTextContent('Body Content');
  });
  test('should render only header if only header provided', () => {
    render(
      <Drawer
        isOpen
        header={<div data-testid="drawer-header">Header Content</div>}
        onClose={() => {}}
      />,
    );
    expect(screen.getByTestId('drawer-header')).toBeInTheDocument();
    // Check if body and footer are NOT in the document
    expect(screen.queryByTestId('drawer-body')).not.toBeInTheDocument();
    expect(screen.queryByRole('footer')).not.toBeInTheDocument();
  });
});
