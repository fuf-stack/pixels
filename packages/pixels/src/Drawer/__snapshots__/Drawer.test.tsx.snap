// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Drawer Component > Should have correct testIds 1`] = `null`;

exports[`Drawer Component > should render with no children 1`] = `null`;

exports[`Drawer Component > should render with only header 1`] = `null`;

exports[`Story Snapshots > AllBackdrops 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    opaque
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    transparent
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    blur
  </button>
</div>
`;

exports[`Story Snapshots > AllPlacements 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    left
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    right
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    top
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    bottom
  </button>
</div>
`;

exports[`Story Snapshots > AllRadii 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    none
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    sm
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    md
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    lg
  </button>
</div>
`;

exports[`Story Snapshots > AllSizes 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    xs
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    sm
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    md
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    lg
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    xl
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    2xl
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    3xl
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    4xl
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    5xl
  </button>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover mb-2 mr-2"
    data-react-aria-pressable="true"
    tabindex="0"
    type="button"
  >
    full
  </button>
</div>
`;

exports[`Story Snapshots > CardWithDrawer 1`] = `
<div>
  <div
    class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-none rounded-small w-full transition-transform-background motion-reduce:transition-none border-divider border"
    tabindex="-1"
  >
    <div
      class="flex p-3 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-small z-auto text-base font-semibold"
    >
      Behold! A Portal to More Stuff!
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] !transition-none data-[pressed=true]:scale-100 bg-default text-default-foreground data-[hover=true]:opacity-hover ml-auto"
        data-react-aria-pressable="true"
        data-testid="drawer_trigger"
        tabindex="0"
        type="button"
      >
        Unleash the Drawer!
      </button>
    </div>
    <hr
      class="shrink-0 bg-divider border-none w-full h-divider border-divider my-0"
      role="separator"
    />
    <div
      class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
    >
      <div
        class="h-48 w-full"
      >
        This is just a placeholder. But a very handsome placeholder.
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > Content 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] !transition-none data-[pressed=true]:scale-100 bg-default text-default-foreground data-[hover=true]:opacity-hover"
    data-react-aria-pressable="true"
    data-testid="drawer_trigger"
    tabindex="0"
    type="button"
  >
    Reveal the Secrets!
  </button>
</div>
`;

exports[`Story Snapshots > CustomWidth 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] !transition-none data-[pressed=true]:scale-100 bg-default text-default-foreground data-[hover=true]:opacity-hover"
    data-react-aria-pressable="true"
    data-testid="drawer_trigger"
    tabindex="0"
    type="button"
  >
    Reveal the Secrets!
  </button>
</div>
`;

exports[`Story Snapshots > Default 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] !transition-none data-[pressed=true]:scale-100 bg-default text-default-foreground data-[hover=true]:opacity-hover"
    data-react-aria-pressable="true"
    data-testid="drawer_trigger"
    tabindex="0"
    type="button"
  >
    Reveal the Secrets!
  </button>
</div>
`;

exports[`Story Snapshots > DefaultOpen 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] !transition-none data-[pressed=true]:scale-100 bg-default text-default-foreground data-[hover=true]:opacity-hover"
    data-react-aria-pressable="true"
    data-testid="drawer_trigger"
    tabindex="0"
    type="button"
  >
    Reveal the Secrets!
  </button>
</div>
`;

exports[`Story Snapshots > DrawerWithTabs 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] !transition-none data-[pressed=true]:scale-100 bg-default text-default-foreground data-[hover=true]:opacity-hover"
    data-react-aria-pressable="true"
    data-testid="drawer_trigger"
    tabindex="0"
    type="button"
  >
    Reveal the Secrets!
  </button>
</div>
`;

exports[`Story Snapshots > Footer 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] !transition-none data-[pressed=true]:scale-100 bg-default text-default-foreground data-[hover=true]:opacity-hover"
    data-react-aria-pressable="true"
    data-testid="drawer_trigger"
    tabindex="0"
    type="button"
  >
    Reveal the Secrets!
  </button>
</div>
`;

exports[`Story Snapshots > Header 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu cursor-pointer outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] !transition-none data-[pressed=true]:scale-100 bg-default text-default-foreground data-[hover=true]:opacity-hover"
    data-react-aria-pressable="true"
    data-testid="drawer_trigger"
    tabindex="0"
    type="button"
  >
    Reveal the Secrets!
  </button>
</div>
`;
